db/db_impl/db_impl_open.cc

```diff
@@ -1564,11 +1585,26 @@ Status DBImpl::Open(const DBOptions& db_options, const std::string& dbname,
         std::string file_path = path + "/" + file_name;
         if (ParseFileName(file_name, &file_number, &file_type) &&
             file_type == kTableFile) {
-          sfm->OnAddFile(file_path);
+          if (known_file_sizes.count(file_name)) {
+            // We're assuming that each sst file name exists in at most one of
+            // the paths.
+            sfm->OnAddFile(file_path, /* compaction */ false,
+                           known_file_sizes.at(file_name));
+            known_file_sizes.erase(file_name);
+          } else {
+            sfm->OnAddFile(file_path);
+          }
         }
       }
     }
 
+    for (const auto& kv : known_file_sizes) {
+      ROCKS_LOG_WARN(impl->immutable_db_options_.info_log,
```

Comment:
This might be a misleading message. Since the calls to sst_file_manager are happening outside the DB mutex, a compaction happening in parallel could have deleted files after we retrieved the list of live files from the current version. In any case, missing files would have been caught by the consistency check.

Reply:
This is mostly a sanity check that this code handles file names correctly. E.g. if I missed that I need to strip the leading '/' from `md.name` a few lines above, that could go unnoticed for a long time without this warning.